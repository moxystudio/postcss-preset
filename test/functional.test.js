'use strict';

const path = require('path');
const fs = require('fs');
const postcss = require('postcss');
const preset = require('..');

const postcssInstance = (() => {
    const { plugins } = preset({
        url: { url: 'inline' },
    });

    const instance = postcss();

    const pluginsEntries = Object.entries(plugins);

    pluginsEntries.forEach(([path, options]) => {
        instance.use(require(path)(options));
    });

    return instance;
})();

const getOutput = (file) => {
    const input = fs.readFileSync(file);

    return postcssInstance.process(input, { from: file });
};

const fixtureFiles = [
    'fixtures/example-1.css',
];

fixtureFiles.forEach((fixtureFile) => {
    it(`should generate correct output for ${fixtureFile}`, async () => {
        const output = await getOutput(path.join(__dirname, fixtureFile));

        expect(output.css).toMatchSnapshot();
    });
});
