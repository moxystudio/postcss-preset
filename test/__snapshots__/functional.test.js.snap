// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate correct output for fixtures/example-1.css 1`] = `
".image {
    position: absolute;
        top: 20px;
        right: 10px;
    font-family: Verdana;

    -webkit-transition: all 0.2s ease;
    transition: all 0.2s ease;
    color: red;
    background: url(\\"data:image/png;base64,\\");
    background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
    background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
    padding: 40px;
    margin-top: calc(10px + 10vh);
        border-color: blue
}



@media (min-width: 60px) {



    .image {
    padding: 60px
    }
}



@media (min-width: 60px) {



    .image {
        top: 30px
    }
}



.image .bar a {
            font-family: Verdana;

            color: red;
            background: url(\\"data:image/png;base64,\\");
            background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
            background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
            padding: 40px;
                border-color: blue
        }



@media (min-width: 60px) {



    .image .bar a {
            padding: 60px
    }
}



.image .bar .link {
            font-family: Verdana;

            color: red;
            background: url(\\"data:image/png;base64,\\");
            background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
            background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
            padding: 40px;
                border-color: blue
        }



@media (min-width: 60px) {



    .image .bar .link {
            padding: 60px
    }
}



.image .bar a:hover {
                font-family: Verdana;

                color: red;
                background: url(\\"data:image/png;base64,\\");
                background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                padding: 40px;
                    border-color: blue
            }



@media (min-width: 60px) {



    .image .bar a:hover {
                padding: 60px
    }
}



.image .bar .link:hover {
                font-family: Verdana;

                color: red;
                background: url(\\"data:image/png;base64,\\");
                background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                padding: 40px;
                    border-color: blue
            }



@media (min-width: 60px) {



    .image .bar .link:hover {
                padding: 60px
    }
}



.buttonWrapper .button.tertiary:enabled:hover:not(.loading) span {
                    color: rgba(0, 0, 0, 0.8);
                }



[dir=\\"ltr\\"] .dirTest {
        margin-left: 10px;
    }



[dir=\\"rtl\\"] .dirTest {
        margin-right: 10px;
    }
"
`;
